src/localconstraints/local_forbidden_sequence.jl

```diff
@@ -0,0 +1,145 @@
+"""
+    LocalForbiddenSequence <: AbstractLocalConstraint
+
+Forbids the given `sequence` of rule nodes ending at the node at the `path`.
+If any of the rules in `ignore_if` appears in the sequence, the constraint is ignored.
+"""
+struct LocalForbiddenSequence <: AbstractLocalConstraint
+    path::Vector{Int}
+    sequence::Vector{Int}
+    ignore_if::Vector{Int}
+end
+
+"""
+    shouldschedule(::Solver, constraint::LocalForbiddenSequence, path::Vector{Int})::Bool
+
+Return true iff the manipulation happened at or above the constraint path.
+"""
+function shouldschedule(::Solver, constraint::LocalForbiddenSequence, path::Vector{Int})::Bool
+    return (length(constraint.path) >= length(path)) && (path== constraint.path[1:length(path)] )
+end
+
+"""
+    function propagate!(solver::Solver, c::LocalForbiddenSequence)
+
+"""
+function propagate!(solver::Solver, c::LocalForbiddenSequence)
+    #TODO: it would be better to precompute the list of `nodes` on the path.
+    # however, in the GenericSolver, uniform holes might be replaced with a rule node, so the instances might change
```

Comment:
Let's try to keep TODOs in issues so that we can keep track of them easier.

Reply:
I will go over and delete all of my TODOs and make sure they are represented by issues.

Reply:
Update: all of my TODOs have been converted to issues.

Reply:
Awesome, thanks!